resolver/resolver.cc

```diff
@@ -661,31 +661,63 @@ class ResolveConstantsWalk {
             return;
         }
 
-        if (send->args.empty()) {
-            return; // The arity mismatch error will be emitted later by infer.
-        }
-
-        for (auto &arg : send->args) {
-            auto argLoc = core::Loc(todo.file, arg.loc());
-            auto *id = ast::cast_tree<ast::ConstantLit>(arg);
+        auto *block = ast::cast_tree<ast::Block>(send->block);
 
-            if (id == nullptr || !id->symbol.exists() || !id->symbol.data(gs)->isClassOrModule()) {
-                if (auto e = gs.beginError(argLoc, core::errors::Resolver::InvalidRequiredAncestor)) {
-                    e.setHeader("Argument to `{}` must be statically resolvable to a class or a module",
-                                send->fun.show(gs));
+        if (!send->args.empty()) {
+            if (auto e = gs.beginError(loc, core::errors::Resolver::InvalidRequiredAncestor)) {
+                e.setHeader("`{}` only accepts a block", send->fun.show(gs));
+
+                string replacement = "";
+                int index = 0;
+                for (auto &arg : send->args) {
+                    auto argLoc = core::Loc(todo.file, arg.loc());
+                    string argString = argLoc.source(gs).value();
+                    if (index > 0) {
+                        auto locDetail = core::Loc::offset2Pos(todo.file.data(gs), send->loc.beginPos());
+                        for (int i = 1; i < locDetail.column; i++) {
+                            replacement += " ";
+                        }
+                    }
+                    replacement += send->fun.show(gs) + " { " + argString + " }";
```

Comment:
Is there a way to use `fmt` with a string that contains literal curly braces `{ ... }`?

Reply:
From https://fmt.dev/latest/syntax.html :
> If you need to include a brace character in the literal text, it can be escaped by doubling: {{ and }}